### ISSUE TITLE
Improve AppSync logging

### ISSUE THREAD

**Issue by tryWabbit** [awslabs/aws-mobile-appsync-sdk-ios#145]
Hello AWS Appsync Team!

Do we have a way to log the current parameters of a Mutation/Query request by a AWSClient object or anyhow? Currently i have to write method to log all the queries so it'll would be great if we can do that on all the requests to make debugging easy.

Thanks and Regards
Kuldeep Tanwar

**Comment by palpatim** [comment]
Hi @dreamBegin,

tl;dr: Unfortunately, AppSync doesn't have consistent logging coverage, so there's not a currently supported way to retrieve that info from the logs.

**Discussion**

The AWSAppSync module uses the `AppSyncLog` module to output log messages to the configured log.  Clients can set up shared logging (which is used by all AWS SDK components, including AppSync, Auth, etc) by doing something similar to the [docs example](https://aws-amplify.github.io/docs/ios/manualsetup#logging): 

```swift
AWSDDLog.sharedInstance.logLevel = .verbose
AWSDDLog.add(AWSDDTTYLogger.sharedInstance)
```

That will cause all AWS SDK modules to log at level `verbose`...assuming they're configured to do so.

Currently, the AWSAppSync module only uses the shared logger in a [few locations](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/search?q=AppSyncLog&unscoped_q=AppSyncLog), and query logging isn't included in that.

I am going to re-title this issue and label it as a feature request to consider for future update.

Note: We recommend setting the log level to `.off` before shipping to production.

**Comment by atierian** [comment]
Thank you for opening this issue. AWS AppSync SDK for iOS entered maintenance mode in September 2023 and will receive no further updates as of September 2024.

Please use Amplify Swift going forward. For information on upgrading to Amplify Swift, refer to the [Upgrade from AppSync SDK documentation](https://docs.amplify.aws/swift/build-a-backend/graphqlapi/upgrade-guide/).
